### ISSUE TITLE
Lore doesn't preserve <!DOCTYPE> declarations in templates in HTML output

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7444]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| spiv reported|
|-|-|
|Trac ID|trac#113|
|Type|defect|
|Created|2003-07-21 12:27:36Z|

<details><summary>Searchable metadata</summary>

```
trac-id__113 113
type__defect defect
reporter__spiv spiv
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__lore lore
keywords__ 
time__1058790456000000 1058790456000000
changetime__1060084910000000 1060084910000000
version__ 
owner__ 
cc__spiv cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
This is possibly a microdom bug rather than a Lore bug, but
I'd have to investigate a little deeper to be sure.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
given that the output always uses a specific doctype
(xhtml?), is this a good idea? or doesn't it?

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
[Clarified the title to make it clear I mean the template's
doctype, not the source document's]

Hmm.  I suppose the output plugin could know what doctype to
use, and be mostly right.  But it's possible to have a
document that is both valid HTML 4.01 and XHTML 1.1 iirc,
and I think allowing the template to set the doctype makes
the most sense.  After all, the template is already where
you set the basic layout, any &lt;meta> tags, stylesheets and
so on, so I think people will expect they can control the
doctype from the template too.

Similarly, for LaTeX output, I intend to make it use
templates as well -- that would allow e.g.
    lore -olatex --config template=stdA4.tex
or
    lore -olatex --config template=USLetter_blueBorders.tex
or even
    lore -olatex --config template=slides.tex
would be possible (i.e. templates would be arbitrary LaTeX
gunk with at least a \documentclass, but possibly much
more).  I think choosing between different paper sizes and
documentclasses in LaTeX is pretty similar to choosing
between different stylesheets, doctypes and whatnot in HTML,
so they should work in a similar way.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
Is this what's causing the generated HOWTO's to ignore the
stylesheet?

It'd be v. nice to have this fixed.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
This is actually a bug in microdom. Microdom doesn't
preserve doctype OR namespace when cloning nodes.

This means that stylesheet display is borked when view the
generated .xhtml files locally in Mozilla

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
Fixed in CVS, revision 1.97 of twisted/web/microdom.py
Tests included.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
Fixed in CVS, revision 1.97 of twisted/web/microdom.py
Tests included.

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
